extensions-core/avro-extensions/src/main/java/org/apache/druid/data/input/avro/GenericAvroJsonProvider.java

```diff
@@ -137,7 +138,15 @@ public void setArrayIndex(final Object o, final int i, final Object o1)
       throw new UnsupportedOperationException();
     }
   }
-
+  
+  private Object transformAvroMapKey(String keyToTransform, Map avroMapObj) {
+    Object key = avroMapObj.keySet().iterator().next();
```

Comment:
Will this work if the map is empty? 

It might be more straightforward to just do something like this in `getMapValue`

```
...
} else if (o instanceof Map) {
  final Map theMap = (Map) o;
  if (theMap.containsKey(s)) {
    return theMap.get(s);
  } else {
    final Utf8 utf8Key = new Utf8(s);
    return theMap.get(utf8Key);
  }
}
...
```

Reply:
your suggestion is more elegant, I will commit it. Thks